### PR TITLE
feat(schema): add intra process comms field to component schema

### PR DESCRIPTION
**Pull request by domire8**
## Description

This PR adds intra process comms field to components in the application schema. I don't know if we should directly go for 2-0-3 or if it might be helpful to have a draft 2-0-3.

## Review guidelines

Estimated Time of Review: 5 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

File: schemas/applications/schema/components.schema.json [hunk 1]
```diff
@@ -138,6 +144,9 @@
             }
           }
         }
+      },
+      {
+
```
Comment (domire8) [inline schemas/applications/schema/components.schema.json:149]:
```suggestion
```

**Comment by domire8** [comment]
Are we interested in adding the `intra_process_communication` field to components or rather not?

File: schemas/applications/schema/application.schema.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "$id": "/2-0-2/application.schema.json",
+  "$id": "/2-0-3-draft/application.schema.json",
```
Comment (domire8) [inline schemas/applications/schema/application.schema.json:2]:
```suggestion
  "$id": "/2-0-2/application.schema.json",
```

**Comment by domire8** [comment]
Not sure what will happen here in the CI, but it seems that there will be a PR generated automatically even though there is no version change..

File: schemas/applications/schema/components.schema.json [hunk 3]
```diff
@@ -79,6 +79,12 @@
             }
           }
         },
+        "intra_process_communications": {
+          "title": "Use Intra-process Communications",
+          "description": "If true, messages on topics which are published and subscribed to within this context will go through a special intra-process communication code code path which can avoid serialization and deserialization, unnecessary copies, and achieve lower latencies in some cases.",
```
Comment (eeberhard) [inline schemas/applications/schema/components.schema.json:84]:
```suggestion
          "title": "Use Intra-Process Communications",
          "description": "If true, publishers and subscribers within the same process will enable intra-process communication which can avoid (de)serialization and unnecessary copies of messages, achieving lower latencies CPU load in some cases.",
```

File: schemas/applications/schema/components.schema.json [hunk 4]
```diff
@@ -79,6 +79,12 @@
             }
           }
         },
+        "intra_process_communications": {
+          "title": "Use Intra-Process Communications",
+          "description": "If true, publishers and subscribers within the same process will enable intra-process communication which can avoid (de)serialization and unnecessary copies of messages, achieving lower latencies CPU load in some cases.",
```
Comment (eeberhard) [inline schemas/applications/schema/components.schema.json:84]: oops
```suggestion
          "description": "If true, publishers and subscribers within the same process will enable intra-process communication which can avoid (de)serialization and unnecessary copies of messages, achieving lower latencies and CPU load in some cases.",
```

**Review by eeberhard — APPROVED** [review]
Great to have it there as default true in case anyone wants to disable it for some reason, and since it's not urgent we can wait to change the version with the next change. Thanks!

**Pull request merged**